### PR TITLE
[Testing] Enable mainnet test utility work with common rpc node

### PR DESCRIPTION
**Pull request by noel2004**
Currently the mainnet test utility is expected to work with a custom geth node as the rpc call provider. If we plan to make it work under any rpc node from common provider (like [quicknode](https://www.quicknode.com/)) there are two issues:

1. A common rpc node do not support the custom "muxTracer" used in the rpc module of bus-mapping
2. Most rpc node provide their trace/debug rpc from a erigon node rather than geth. Some parameters is different and there is still no `refund` field is included in the defaut trace data.

This PR has induced a `fix-refund` feature into mainnet testing utility. With this feature it would:

+ unify the arguments of debug rpc call, so it is compatible with both geth and erigon
+ use common avaliable tracer instead of the custom one
+ rebuid the refund field while handing tx instead of read it from the structural log inside geth trace object

### CODE REVIEW

**Comment by lispc** [comment]
I will review more carefully later but some ideas now:  1 should the "legacy" word be replaced by "erigon"? 2 usually we don't want too many feature flags unless really needed. For example i plan to remove the "shanghai" feature soon since it is outdated. And i also doubt about features like "enable-storage".  If new logic is strictly better/supetset of old logics(i don't know just "if"), we should overwrite instead of keeping both.

File: bus-mapping/src/evm/opcodes/sstore.rs [hunk 1]
```diff
@@ -109,12 +109,69 @@ impl Opcode for Sstore {
...
+                        // reset to original inexistent slot (2.2.2.1)
+                        tx_refund_new +=
+                            GasCost::SSTORE_SET.as_u64() - GasCost::WARM_ACCESS.as_u64();
+                    } else {
+                        // reset to original existing slot (2.2.2.2)
+                        tx_refund_new +=
+                            GasCost::SSTORE_RESET.as_u64() - GasCost::WARM_ACCESS.as_u64();
+                    }
+                }
+            }
```
Comment (lispc) [inline bus-mapping/src/evm/opcodes/sstore.rs]: line 120 to line 146 is copied from some where?  zkevm-circuits dir?  better don't repeat..

**Comment by lispc** [comment]
> I will review more carefully later but some ideas now: 1 should the "legacy" word be replaced by "erigon"? 2 usually we don't want too many feature flags unless really needed. For example i plan to remove the "shanghai" feature soon since it is outdated. And i also doubt about features like "enable-storage". If new logic is strictly better/supetset of old logics(i don't know just "if"), we should overwrite instead of keeping both.

seems ok.

**Review by lispc — APPROVED** [review]

File: bus-mapping/src/evm/opcodes/sstore.rs (hunk 1 above)
Comment (noel2004) [reply bus-mapping/src/evm/opcodes/sstore.rs]: has move the `calc_expected_tx_refund` into bus-mapping and both bus-mapping and zkevm-circuit use it

**Pull request merged**